metals/src/main/scala/scala/meta/internal/builds/SbtBuildTool.scala

```diff
@@ -85,13 +87,50 @@ case class SbtBuildTool(
   private def writeSbtMetalsPlugin(
       workspace: AbsolutePath
   ): Unit = {
+
+    def sbtMetaDirs(
+        dir: AbsolutePath,
+        parentHasSbtSrc: Boolean,
+        acc: List[AbsolutePath]
+    ): List[AbsolutePath] = {
+
+      val next = dir.resolve("project")
+      if (dir.exists) {
+        val (hasScalaSrc, hasSbtSrc) = dir.list.foldLeft((false, false)) {
```

Comment:
Maybe this could be easier to udnerstand:
```scala
val files = dir.list
val hasScalaSrc = files.exists(_.isScala)
val hasSbtSrc = files.exists(f => f.isSbt && f.filename != "metals.sbt" )
```
Granted it's not all in one go, but this list is rather small.